pricegraph/src/lib.rs

```diff
@@ -41,3 +44,123 @@ pub struct TransitiveOrder {
     /// The effective sell amount for this transient order.
     pub sell: f64,
 }
+
+impl TransitiveOrder {
+    /// Retrieves the exchange price for this order.
+    pub fn price(&self) -> f64 {
+        self.buy / self.sell
+    }
+
+    /// Retrieves the effective price for this order after fees are condidered.
+    ///
+    /// Note that `effective_price > price`.
+    pub fn effective_price(&self) -> f64 {
```

Comment:
Thinking about the naming (I'm fine with the way it is). Is this the "matchable" price, e.g. the price at which someone could take this order? If so maybe we can call it matcheable or takeable price?

Reply:
No, this is the effective limit price, that is, the minimum exchange rate that can be used in a solution because of fees. If the exchange rate were lower, than the solution would not be accepted because the executed_buy_amount / executed_sell_amount would be lower than the order limit price.

Reply:
I think for the orderbook visualization, we should probably use this value, or else we might see a small amount of overlap that is not actually there once fees are factored in.

Reply:
For the order to be "matchable", you need to place an order whose effective exchange rate overlaps with this order's effective exchange rate.

Reply:
speaking of which, I renamed price to exchange rate :tada:. Lets try and keep this top level API using the language proposed by @marcovc.